backend/src/openbeheer/api/drf_spectacular/schema.py

```diff
@@ -21,7 +21,13 @@ class MsgSpecExtension(OpenApiSerializerExtension):
     def _matches(cls, target: Any) -> bool:
         # For generic types
         if get_origin(target):
-            return issubclass(get_origin(target), SUPPORTED_MSG_CLASSES)
+            if issubclass(get_origin(target), SUPPORTED_MSG_CLASSES):
+                return True
+
+            # Example: list[OBOption] origin is list, and not in SUPPORTED_MSG_CLASSES,
+            # but the argument OBOption is a Struct, so it needs to use this extension.
+            return any(cls._matches(arg) for arg in get_args(target))
```

Comment:
Should we even recurse?
And `SUPPORTED_MSG_CLASSES` for schema generation is different from encode/decode?
Could we just
```python
try:
    return bool(schema_components(target))
except:
    return False
```
or are we guarding for bad output, not for exceptions?

Reply:
I like it! Much simpler. I need to investigate why it fails for `ZGWResponse[ZaakTypeSummary]` with this error:
```
drf_spectacular.plumbing.UnableToProceedError: openbeheer.types._zgw.ZGWResponse[openbeheer.zaaktype.api.views.ZaakTypeSummary]
```